### DISCUSSION TITLE
[SUPPORT]: getting yaml as a single line string

### DISCUSSION THREAD

**Discussion by rayzorben** [mikefarah/yq#1671] [Q&A] [answered]
I am moving from another yq library which allows you to read an array such as .disk[] as a single line string. This allows you to do the pattern:

```
while read disk; do
  echo $disk | yq '.device' # retrieves the .device value of the current .disk[]
done< <(yq -c '.disks[]' my.yml')
```

The -c would keep it as a single line and be able to parse it, so essentially it was one line per 'disk' in the disks array.

Does go-yq have something similar?

**Comment by mikefarah** [comment, marked as answer]
There's a similar question I answered in stackoverflow here: https://stackoverflow.com/questions/62898925/using-yq-in-for-loop-bash/70162342#70162342 - does that help? I can't see the my.yml file, but it does look like the command you have will work in `yq` - just drop the `-c`.

**Comment by rayzorben** [comment]
The `-o=j -I=0` worked wonderfully, thanks.
